player/audio.c

```diff
@@ -55,14 +55,16 @@ static void update_speed_filters(struct MPContext *mpctx)
         return;
 
     double speed = mpctx->opts->playback_speed;
-    double resample = mpctx->speed_factor_a;
+    double resample = mpctx->speed_factor_a * mpctx->opts->playback_pitch;
     double drop = 1.0;
 
     if (!mpctx->opts->pitch_correction) {
         resample *= speed;
         speed = 1.0;
     }
 
+    speed /= mpctx->opts->playback_pitch;
+
     if (mpctx->display_sync_active) {
```

Comment:
Is this expected this new feature will not work correctly with -adrop and -tempo sync modes?